python/grass/jupyter/timeseries.py

```diff
@@ -0,0 +1,435 @@
+# MODULE:    grass.jupyter.timeseries
+#
+# AUTHOR(S): Caitlin Haedrich <caitlin DOT haedrich AT gmail>
+#
+# PURPOSE:   This module contains functions for visualizing raster and vector
+#            space-time datasets in Jupyter Notebooks
+#
+# COPYRIGHT: (C) 2021 Caitlin Haedrich, and by the GRASS Development Team
+#
+#           This program is free software under the GNU General Public
+#           License (>=v2). Read the file COPYING that comes with GRASS
+#           for details.
+"""Create and display visualizations for space-time datasets."""
+
+import tempfile
+import os
+import weakref
+import shutil
+import grass.script as gs
+
+from .display import GrassRenderer
+from .region import RegionManagerForTimeSeries
+
+
+def fill_none_values(names):
+    """Replace `None` values in array with previous item"""
+    for i, name in enumerate(names):
+        if name == "None":
+            names[i] = names[i - 1]
+        else:
+            pass
+    return names
+
+
+def collect_layers(timeseries, element_type, fill_gaps):
+    """Create lists of layer names and start_times for a
+    space-time raster or vector dataset.
+
+    For datasets with variable time steps, makes step regular with
+    "gran" method for t.rast.list or t.vect.list then fills in
+    missing layers with previous time step layer.
+
+    :param str timeseries: name of space-time dataset
+    :param str element_type: element type, "stvds" or "strds"
+    :param bool fill_gaps: fill empty time steps with data from previous step
+    """
+    # NEW WAY: Comment in after PR 2258 is merged
```

Comment:
Note that #2258 is only for _t.rast.list_, the module level code from _t.rast.list_ needs to be ported to _t.vect.list_ (the library code is the shared).

Reply:
Right! Should I wait for that to be ported? It seems confusing and long to keep one method for t.rast.list and another for t.vect.list.

Reply:
Just keep it as is, only change the comment. It works as is, so I probably won't change t.vect.list in time for 8.2.